std/encoding.d

```diff
@@ -3297,6 +3298,30 @@ class EncodingSchemeUtf32Native : EncodingScheme
     assert(ub.length == 8);
 }
 
+// this hack allows us to register all the classes without having dependencies
+// on other modules. The function is eterned, and then called from phobosinit.
+// Note that EncodingScheme.register uses the default constructor of each of
+// these classes via the Object.factory method, and then calls the names()
+// function. None of these functions above depend on external modules, so we
+// are safe to call this from an external module without risk of creating
+// cycles.
+extern(C) void std_encoding_shared_static_this()
```

Comment:
No. `EncodingScheme.register` allocates, so this can't be marked `@nogc`. In any case, it's called before `main`, so I don't see why it needs to be.